dali/kernels/reduce/reduce_gpu.cu

```diff
@@ -58,6 +58,100 @@ template class SumGPU<float, int32_t>;
 template class SumGPU<uint64_t, uint32_t>;
 template class SumGPU<float, uint32_t>;
 template class SumGPU<float, float>;
+template class SumGPU<int32_t, int32_t>;
+template class SumGPU<int16_t, int16_t>;
+template class SumGPU<uint16_t, uint16_t>;
+template class SumGPU<uint8_t, uint8_t>;
+
+
```

Comment:
One crazy idea, if we plan to have many reductions that follow the same pattern we could have a:
```
REDUCTION_IMPL(Kernel, Impl)  \
  template <typename Out, typename In> \
  class Kernel<Out, In>::Impl : public Impl<Out, In> { \
  
  ...
```

Basically it'd cover all the repeated code and later you just do:
```
REDUCTION_IMPL(MinGPU, reduce_impl::MinImplGPU);
REDUCTION_IMPL(MaxGPU, reduce_impl::MaxImplGPU);
...
```
I have mixed feeling about having so much inside a template, but on the other hand, now there's a lot of boiler-plate here. Second opinion maybe? @mzient ?